core/dbt/deprecations.py

```diff
@@ -35,6 +35,17 @@ def show(self, *args, **kwargs) -> None:
             active_deprecations.add(self.name)
 
 
+class ExposureNameDeprecation(DBTDeprecation):
+    _name = "exposure-name"
+    # todo: word this better - it's bad.  Also do we want to set a timeline for when
+    # this will become an error?
+    _description = """\
+    Exposure names may not contain spaces and can only contain letters, numbers and underscores.
+    {exposure} does not follow this pattern.  Please use the label attribute for more complex
+    names.
+    """
```

Comment:
@jtcohen6 please provide the promised wording 😄

Reply:
How about this:
```python
    _description = """\
    Starting in v1.3, the 'name' property of exposures should contain only contain letters, numbers, and underscores.
    Exposures support a new property, 'label', that can contain spaces, capital letters, and special characters.
    {exposure} does not follow this pattern.
    Please update its 'name', and use the 'label' property for its human-friendly title.
    This will raise an exception in a future version of 'dbt-core'.
    """
```
In the case of metrics, we'd explicitly documented the metric spec as "experimental," and reserved the right to make breaking changes. In that context, it's nice for us to provide backward compatibility for one minor version, and reasonable to expect a quick turnaround (breaking change by v1.4).

Exposures have been around for longer, and without an "experimental" denotation — so I don't think we should commit to a quick turnaround for actual deprecation. The soonest I could see doing this is if we significantly expand the capabilities of exposures, as part of a potential initiative next year. Otherwise, in dbt v2.0.

Given that, I also don't feel the need to open an issue to track this. We'll have an opportunity to revisit this either when we reinvest in exposures, or when we revisit all of our deprecations ahead of v2 (just as we did ahead of v1).